### ISSUE TITLE
Relu for S2Activation not acting as expected. Why isn't the left graphic only blue?

### ISSUE THREAD

**Issue by blondegeek** [e3nn/e3nn#123]
```
import torch
import e3nn
from e3nn.tensor.spherical_tensor import SphericalTensor
from e3nn.non_linearities.s2 import S2Activation
import plotly
import plotly.graph_objects as go
from plotly.subplots import make_subplots

L_max = 6
Rs_in = [(1, L) for L in range(L_max + 1)]

s2act = S2Activation(Rs_in, torch.relu, 300)

tetra_coords = torch.tensor( # The easiest way to construct a tetrahedron is using opposite corners of a box
    [[0., 0., 0.], [1., 1., 0.], [1., 0., 1.], [0., 1., 1.]]
)
tetra_coords -= tetra_coords.mean(-2)
tetrahedron = SphericalTensor.from_geometry(tetra_coords, lmax=6).signal

input = tetrahedron
output = s2act(input)

fig = make_subplots(cols=cols, rows=rows, specs=specs)

r1, f1 = SphericalTensor(input).plot(radius=False)  # leave off last 4 scalars used for gate
r2, f2 = SphericalTensor(output).plot(radius=False)
fmax = torch.max(f1.abs().max(), f2.abs().max()).item()
fig.add_trace(go.Surface(x=r1[..., 0], y=r1[..., 1], z=r1[..., 2], surfacecolor=f1, 
                         cmin=-fmax, cmax=fmax, colorscale='RdBu'), 
              row=1, col=1)
fig.add_trace(go.Surface(x=r2[..., 0], y=r2[..., 1], z=r2[..., 2], surfacecolor=f2, 
                         cmin=-fmax, cmax=fmax, colorscale='RdBu'), 
              row=1, col=2)

fig.show()
```
![operations_graphics_-_Jupyter_Notebook](https://user-images.githubusercontent.com/4080470/89252642-2e4d8400-d5cf-11ea-8abb-01709bd45a4c.jpg)

In contrast, if I replace relu with sigmoid...
![operations_graphics_-_Jupyter_Notebook](https://user-images.githubusercontent.com/4080470/89252676-4d4c1600-d5cf-11ea-9474-01637bd4ada2.jpg)

**Comment by mariogeiger** [comment]
It’s surprising that the two plots are so similar. The usual reason for why relu behave differently than sigmoid is that it is a non smooth function. Relu introduce very high frequencies into the signal, and truncating them can usually give negative values. Try with a higher L to see

**Comment by mariogeiger** [comment]
Oh but your input signal is positive! This a why relu does nothing of course

**Comment by blondegeek** [comment]
I'm confused by what you mean by the input signal being positive

For `relu` (`f1` is before act and `f2` is after)
```
f1.min(), f2.min()
>> (tensor(-0.2206), tensor(-0.2140))
```

Versus with `sigmoid`
```
f1.min(), f2.min()
>> (tensor(-0.2206), tensor(0.3186))
```

**Comment by mariogeiger** [comment]
I mean that the signal before truncating with the first 6 spherical harmonics is a sum of dirac deltas. So it's a positive signal.
I does not surprise me at all that once projected you get slight negative values both times:
- after the projection of the deltas
- after the second projection after the activation function

**Comment by blondegeek** [comment]
Ah! I see what you mean -- you get "negative" artifacts regardless. That
makes sense. Thanks!
